packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -263,26 +379,42 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
 
   // When the value prop changes, update the selectedIndex
   React.useEffect(() => {
-    dispatch([
-      'select',
-      options.findIndex((option) => option.value === valueProp),
-    ]);
-  }, [options, valueProp]);
-
-  // Call user-defined onChange when the value actually changes
-  React.useEffect(() => {
-    // Prevent user-defined onChange to be called on mount
-    if (!mounted.current) {
-      mounted.current = true;
-      return;
+    if (isMultipleEnabled(valueProp, multiple) && !!valueProp) {
```

Comment:
I think this useEffect might need an early return if the `valueProp` is the same as what is currently in the state. We were doing something similar previously.

Reply:
Agree. It will be interesting to add check if two arrays are the same. Do we care about order of items in array?

Reply:
I think order shouldn't matter, but the user should not change the order anyway, so it's also fine if we execute this when the order is different.